src/ADAL.PCL.Android/BrokerHelper.cs

```diff
@@ -0,0 +1,185 @@
+//----------------------------------------------------------------------
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved
+// Apache License 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Android.Accounts;
+using Android.App;
+using Android.Content;
+using Java.IO;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    class BrokerHelper : IBrokerHelper
+    {
+        private static SemaphoreSlim readyForResponse = null;
+        private static AuthenticationResultEx resultEx = null;
+
+        private BrokerProxy mBrokerProxy = new BrokerProxy(Application.Context);
+
+        public IPlatformParameters PlatformParameters { get; set; }
+
+        private bool WillSkipBroker()
+        {
+            PlatformParameters pp = PlatformParameters as PlatformParameters;
+            if (pp != null)
+            {
+                return pp.SkipBroker;
+            }
+
+            return true;
+        }
+
+        public bool CanInvokeBroker { get { return !WillSkipBroker() && mBrokerProxy.CanSwitchToBroker(); } }
+
+
+        public async Task<AuthenticationResultEx> AcquireTokenUsingBroker(IDictionary<string, string> brokerPayload)
+        {
+            resultEx = null;
+            readyForResponse = new SemaphoreSlim(0);
+            try
+            {
+                await Task.Run(() => AcquireToken(brokerPayload));
+            }
+            catch (Exception exc)
+            {
+                PlatformPlugin.Logger.Error(null, exc);
+                throw;
+            }
+            await readyForResponse.WaitAsync();
+            return resultEx;
+        }
+        public void AcquireToken(IDictionary<string, string> brokerPayload)
+        {
+            Context mContext = Application.Context;
+            AuthenticationRequest request = new AuthenticationRequest(brokerPayload);
+            PlatformParameters platformParams = PlatformParameters as PlatformParameters;
+
+            // BROKER flow intercepts here
+            // cache and refresh call happens through the authenticator service
+            if (mBrokerProxy.VerifyUser(request.LoginHint,
```

Comment:
can switch to broker check happens in AcquireTokenHandlerBase as it needs to apply for all platforms